src/zenml/artifact_stores/base_artifact_store.py

```diff
@@ -403,6 +456,34 @@ def _register(self) -> None:
 
         default_filesystem_registry.register(filesystem_class)
 
+    def __init_subclass__(cls) -> None:
+        """Wrap all abstract methods implementations with a path verifier.
+
+        This is done to ensure that the path is within the artifact store
+        bounds.
+        """
+        for abc_method in inspect.getmembers(BaseArtifactStore):
+            if getattr(abc_method[1], "__isabstractmethod__", False):
+                setattr(
+                    cls, abc_method[0], _verify(getattr(cls, abc_method[0]))
+                )
+
+    def _inner_verify(self, name: PathType) -> None:
+        """Verify that the path is inside artifact store bounds.
+
+        Args:
+            name: The path to verify.
+
+        Raises:
+            IOError: If the path is not inside the artifact store bounds.
+        """
+        request_path = _sanitize_potential_path(name)
+        if not request_path.startswith(self.path):
+            raise IOError(
```

Comment:
I would raise `FileNotFoundError` here. This way, it is properly caught in the REST API endpoints and reported more accurately to the dashboard and clients. It is also the standard library exception for this sort of thing.

Reply:
done